core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -69,55 +67,51 @@ class TileMap {
         return tileMatrix[arrayXIndex][arrayYIndex]!!
     }
 
-    operator fun get(vector: Vector2): TileInfo {
-        return get(vector.x.toInt(), vector.y.toInt())
-    }
-
-    fun getTilesInDistance(origin: Vector2, distance: Int): List<TileInfo> {
-        val tilesToReturn = mutableListOf<TileInfo>()
-        for (i in 0 .. distance) {
-            tilesToReturn += getTilesAtDistance(origin, i)
-        }
-        return tilesToReturn
-    }
-
-    fun getTilesAtDistance(origin: Vector2, distance: Int): List<TileInfo> {
-        if(distance==0) return listOf(get(origin))
-        val tilesToReturn = ArrayList<TileInfo>()
-
-        fun addIfTileExists(x:Int,y:Int){
-            if(contains(x,y))
-                tilesToReturn += get(x,y)
-        }
-
-        val centerX = origin.x.toInt()
-        val centerY = origin.y.toInt()
-
-        // Start from 6 O'clock point which means (-distance, -distance) away from the center point
-        var currentX = centerX - distance
-        var currentY = centerY - distance
-
-        for (i in 0 until distance) { // From 6 to 8
-            addIfTileExists(currentX,currentY)
-            // We want to get the tile on the other side of the clock,
-            // so if we're at current = origin-delta we want to get to origin+delta.
-            // The simplest way to do this is 2*origin - current = 2*origin- (origin - delta) = origin+delta
-            addIfTileExists(2*centerX - currentX, 2*centerY - currentY)
-            currentX += 1 // we're going upwards to the left, towards 8 o'clock
-        }
-        for (i in 0 until distance) { // 8 to 10
-            addIfTileExists(currentX,currentY)
-            addIfTileExists(2*centerX - currentX, 2*centerY - currentY)
-            currentX += 1
-            currentY += 1 // we're going up the left side of the hexagon so we're going "up" - +1,+1
-        }
-        for (i in 0 until distance) { // 10 to 12
-            addIfTileExists(currentX,currentY)
-            addIfTileExists(2*centerX - currentX, 2*centerY - currentY)
-            currentY += 1 // we're going up the top left side of the hexagon so we're heading "up and to the right"
-        }
-        return tilesToReturn
-    }
+    operator fun get(vector: Vector2): TileInfo = get(vector.x.toInt(), vector.y.toInt())
+
+    fun getTilesInDistance(origin: Vector2, distance: Int): Sequence<TileInfo> =
+            getTilesInDistanceRange(origin, 0..distance)
+
+    fun getTilesInDistanceRange(origin: Vector2, range: IntRange): Sequence<TileInfo> =
+            range.asSequence().flatMap { getTilesAtDistance(origin, it) }
```

Comment:
as mentioned before, this was added for getting a range, which is used by getTilesInDistance anyway.  Sequences are returned as well since most places don't need list (only 3 places as mentioned before)